### ISSUE TITLE
Show simple webpage during update 

### ISSUE THREAD

**Issue by ja-pa** [TheTorProject/lepidopter#95]
There is no indication of ongoing update through web interface. It would be nice to have some simple page which could highlight that something is going on.

**Comment by darkk** [comment]
@ja-pa do you mean lepidopter software update/upgrade or update on network measurement progress?

**Comment by ja-pa** [comment]
I mean lepidopter software update/upgrade.

**Comment by muellermartin** [comment]
I've just installed Lepidoter on a Raspberry Pi B+ and I was wondering why I could not open the web interface although the device was reachable via network. Only after I used SSH to look at the list of processes (to look for something like a web server), I saw that a Python script was writing to `/var/log/ooni/lepidopter-update.log` which indicated that an update was running:

```
1970-01-01 00:01:18,666 - INFO - Checking for update
1970-01-01 00:01:19,932 - ERROR - Failed to learn the latest version
2019-07-12 15:27:15,969 - INFO - Checking for update
2019-07-12 15:27:18,064 - INFO - Updating 8->9
2019-07-12 15:27:34,695 - INFO - Running install script
```

The Raspberry Pi model that I'm using is quite slow, so Python modules which needs compiled parts are taking quite some time to install. I even tried to restart the device once before I noticed an ongoing update. Thus a webpage indicating that an update is in progress, would be really helpful.

**Comment by koushikcs562** [comment]
Hello @ja-pa, @darkk, and @muellermartin,
I agree that this lack of visual feedback during an update is a major pain point, especially on slower hardware like the Raspberry Pi B+. A simple "Update in Progress" page would significantly improve the user experience and prevent accidental reboots.
I'd like to take on implementing this feature.
My Approach
The core challenge is ensuring this maintenance page is served before the main web server restarts or becomes unavailable during the update process.
State Flag: I will introduce a simple, temporary flag file (e.g., /var/run/lepidopter_updating) right at the start of the software update script.
Web Server Logic: I will modify the web server configuration (or the web application's routing logic) to check for the existence of this flag file on every incoming request.
Serve Maintenance Page:
If the flag exists, the web server will immediately serve a static, simple HTML page that says "Lepidopter Update in Progress. Please wait and DO NOT power off the device."
If the flag does not exist, the request proceeds to the normal web interface.
Cleanup: The software update script will be modified to delete the flag file only upon successful completion of the entire update process.
